### PR TITLE
Paralize the StitchBlobs

### PR DESCRIPTION
**Pull request by hongyuchen1030**
Modify the StitchBlobs.cpp 
Also add comments in BlobUtilities.h for some parallel notes

### CODE REVIEW

File: src/blobs/StitchBlobs.cpp [hunk 1]
```diff
@@ -21,6 +21,7 @@
 #include "Constants.h"
 #include "CoordTransforms.h"
 #include "BlobUtilities.h"
+#include <math.h>
```
Comment (hongyuchen1030) [inline src/blobs/StitchBlobs.cpp:24]: replace with cmath

File: src/blobs/StitchBlobs.cpp [hunk 2]
```diff
@@ -109,13 +111,1859 @@ struct Tag {
...
+
+		///	<summary>
+		///		Tag for MPI communication for sending/receiving Tag index
+		///	</summary>
+		int gather_tag_index = 108;
+
+		///	<summary>
+		///		The MPI Communicator
+		///	</summary>
+		MPI_Comm _comm;
```
Comment (hongyuchen1030) [inline src/blobs/StitchBlobs.cpp:138]: use it with m_comm;

File: src/blobs/StitchBlobs.cpp [hunk 3]
```diff
@@ -109,13 +111,1859 @@ struct Tag {
...
+		///	</summary>
+		std::vector<int> vecScatterCounts_index;
+
+	public:
+
+		///	<summary>
+		///		Constructor that will read in std::vector< std::vector<Tag>> vecAllBlobTags and MPI communicator
+		///		It will also create the derived MPI_Datatype for Tag and commit it.
+		///	</summary>	
+		TagCollectiveOP(MPI_Comm communicator, std::vector< std::vector<Tag>> vecAllBlobTags){
```
Comment (hongyuchen1030) [inline src/blobs/StitchBlobs.cpp:236]: different lines for input.
period, 
pass in as reference

File: src/blobs/StitchBlobs.cpp [hunk 4]
```diff
@@ -109,13 +111,1859 @@ struct Tag {
...
+			Tag_displacements[2] = MPI_Aint_diff(Tag_displacements[2], base_address);
+			MPI_Type_create_struct(tagFieldsCount, Tag_block_lengths, Tag_displacements, Tag_typesig, &MPI_Tag_type);
+			try{				
+				MPI_Type_commit(&MPI_Tag_type);
+			} catch (MPI::Exception failure) {
+				_EXCEPTION1("The MPI routine MPI_Type_commit(&MPI_Tag_type) failed: %s.\n", failure.Get_error_string());
+				MPI::COMM_WORLD.Abort(1);	
+			}
+		}
+		///	<summary>
```
Comment (hongyuchen1030) [inline src/blobs/StitchBlobs.cpp:267]: only 1 blank line before any comment summary
1 space after //

File: src/blobs/StitchBlobs.cpp [hunk 5]
```diff
@@ -1403,13 +3360,48 @@ try {
...
-	typedef MapGraph::const_iterator MapGraphConstIterator;
-	typedef MapGraph::iterator MapGraphIterator;
+	//================================Actual Parallization Starts===================================
+	//1. Exchang the vecAllBlobs; vecAllBlobTags; vecPrevBlobBoxesDeg; vecGlobalTimes
+	//==============================================================================================
+
+#if defined(TEMPEST_MPIOMP)
+
+	//We still need the original unexchanged data for these two variables later
+	GlobalTimesExchangeOp MPI_exchangedGlobalTimes(MPI_REAL_COMM,vecGlobalTimes, processorResponsibalForFile_LB, processorResponsibalForFile_UB);//Declare here since it needs to be reverted later
```
Comment (hongyuchen1030) [inline src/blobs/StitchBlobs.cpp:3372]: break into several lines

**Comment by paullric** [comment]
Can you remove modifications to .gitignore in the PR?

**Pull request merged**